doc/developers/advanced_installation.rst

```diff
@@ -22,127 +33,191 @@ basis to help users test bleeding edge features or bug fixes::
 Building from source
 =====================
 
-In the vast majority of cases, building scikit-learn for development purposes
-can be done with::
+.. _git_repo:
 
-    pip install cython pytest flake8
+Use `Git <https://git-scm.com/>`_ to check out the latest source from the
+`scikit-learn repository <https://github.com/scikit-learn/scikit-learn>`_ on
+`Github <https://github.com>`_ and pip to build the project::
 
-Then, in the main repository::
+    git clone git://github.com/scikit-learn/scikit-learn.git
```

Comment:
We typically want contributors to clone from their fork.

Reply:
You can always add another remote later. Personally I prefer to use `ogrisel` as my fork and `origin` for the upstream repo.

Reply:
Maybe just add a comment like "if you plan on submitting a pull-request, you should clone from your fork instead"

Reply:
Fixed.